cdap-ui/app/features/adapters/services/etl-create-factory.js

```diff
@@ -6,12 +6,39 @@ angular.module(PKG.name + '.feature.adapters')
       this.dataSrc = new MyDataSource(scope);
     }
 
+    function getIcon(plugin) {
+      var iconMap = {
+        'script': 'code',
+        'twitter': 'twitter',
+        'cube': 'cubes',
+        'data': 'database',
+        'database': 'database',
+        'table': 'table'
+      };
+
+      var pluginName = plugin.toLowerCase(),
+          icons = Object.keys(iconMap),
+          icon = 'plug';
+      for(var i=0; i<icons.length; i++) {
+        if (pluginName.indexOf(icons[i]) !== -1) {
+          icon = iconMap[icons[i]];
+          break;
+        }
+      }
+      return icon;
+    }
+
     AdapterApiFactory.prototype.fetchSources = function(adapterType) {
       this.dataSrc.request({
         _cdapPath: '/templates/' + adapterType + '/extensions/source'
       })
         .then(function(res) {
           this.scope.defaultSources = res;
+          if (res.length) {
```

Comment:
isn't the res.length redundant?
for forEach won't iterate if it's empty, anyways, right?